BackEnd/src/routes/ProjectRoutes.php

```diff
@@ -0,0 +1,308 @@
+<?php
+use Chumper\Zipper\Zipper;
+use Slim\Http\Response as Response;
+use Slim\Http\Request as Request;
+use Symfony\Component\Filesystem\Filesystem;
+use HostMyDocs\Controllers\ProjectController;
+
+$slim->get('/listProjects', function (Request $request, Response $response)
+{
+    $projects = [];
+    try {
+        $projects = ProjectController::listProjects($this->get('storageRoot'), $this->get('archiveRoot'));
+    } catch (\Exception $e) {
+    }
+
+    $cacheProvider = $this->get('cache');
+    return $cacheProvider->withEtag($response->withJson($projects), md5(json_encode($projects)));
+});
+
+
+$slim->post('/addProject', function (Request $request, Response $response)
+{
+    // increasing execution time
+    ini_set('max_execution_time', 3600);
+
+    $errorMessage = null;
+    $name = null;
+    $version = null;
+    $language = null;
+
+    $filesystem = new Filesystem();
+    $requestParams = $request->getParsedBody();
+
+    if (count($requestParams) === 0) {
+        $errorMessage = 'no parameters found';
+        $response = $response->write($errorMessage);
+        return $response->withStatus(400);
+    }
+
+
+    error_log('Processing a new request');
+
+    if (array_key_exists('name', $requestParams)) {
+        $name = $requestParams['name'];
+    }
+
+    if (array_key_exists('version', $requestParams)) {
+        $version = $requestParams['version'];
+    }
+
+    if (array_key_exists('language', $requestParams)) {
+        $language = $requestParams['language'];
+    }
+
+    $files = $request->getUploadedFiles();
+    $archive = null;
+
+
+    error_log('Checking provided parameters');
+
+    if ($name === null) {
+        $errorMessage = 'name is empty';
+    }
+
+    if (strpos($name, '/') !== false) {
+        $errorMessage = 'name cannot contains slashes';
+    }
+
+    if ($language === null) {
+        $errorMessage = 'language is empty';
+    }
+
+    if (strpos($language, '/') !== false) {
+        $errorMessage = 'language cannot contains slashes';
+    }
+
+    if ((array_key_exists('archive', $files))) {
+        $archive = $files['archive'];
+    } else {
+        $errorMessage = 'no file provided';
+        // this one needed to prevent next if to throw a null pointer exeption
+        $response = $response->write($errorMessage);
+        return $response->withStatus(400);
+    }
+
+    if (pathinfo($archive->getClientFilename(), PATHINFO_EXTENSION) !== 'zip') {
+        $errorMessage = 'archive is not a zip file';
+    }
+
+    if ($errorMessage !== null) {
+        $response = $response->write($errorMessage);
+        return $response->withStatus(400);
+    }
+
+    error_log('Parameters OK');
```

Comment:
You should not use that. 
Use PSR-3 compatible logger (like [monolog](https://github.com/Seldaek/monolog) for example )